src/lib/Cashlink.ts

```diff
@@ -0,0 +1,453 @@
+import { Utf8Tools } from '@nimiq/utils';
+import { NetworkClient, DetailedPlainTransaction } from '@nimiq/network-client';
+import { SignTransactionRequestLayout } from '@nimiq/keyguard-client';
+import { loadNimiq } from './Helpers';
+import { CashlinkState } from './PublicRequestTypes';
+
+export const CashlinkExtraData = {
+    FUNDING:  new Uint8Array([0, 130, 128, 146, 135]), // 'CASH'.split('').map(c => c.charCodeAt(0) + 63)
+    CLAIMING: new Uint8Array([0, 139, 136, 141, 138]), // 'LINK'.split('').map(c => c.charCodeAt(0) + 63)
+};
+
+export interface CashlinkEntry {
+    address: string;
+    keyPair: Uint8Array;
+    value: number;
+    message: string;
+    state: CashlinkState;
+    date: number;
```

Comment:
I think since the `address` is a `string` that's clear. And `value` or `amount` is always in luna until displayed, so also clear. I renamed `date => timestamp`, so it should also be clear that it's in seconds.

Reply:
Well, there are exceptions to adsress as user firnedly string or value in luna.
`timestamp` could still mean Milliseconds (which it typically does in JS) but ok.